modules/phase_field/src/materials/CrossTermBarrierFunctionMaterial.C

```diff
@@ -74,10 +86,13 @@ CrossTermBarrierFunctionMaterial::computeQpProperties()
       switch (_g_order)
       {
         case 0: // SIMPLE
-          _prop_g[_qp]         +=  _W_ij[_num_eta*i + j] * (*_eta[i])[_qp] * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
-          (*_prop_dg[i])[_qp]  +=  2 * _W_ij[_num_eta*i + j] * 2 * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
-          (*_prop_d2g[i])[_qp] +=  2 * _W_ij[_num_eta*i + j] * 2 * (*_eta[j])[_qp] * (*_eta[j])[_qp];
-          break;
+          _prop_g[_qp]         +=  _W_ij[_num_eta * i + j] * (*_eta[i])[_qp] * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
+          (*_prop_dg[i])[_qp]  +=  _W_ij[_num_eta * i + j] * 2 * (*_eta[i])[_qp] * (*_eta[j])[_qp] * (*_eta[j])[_qp];
+          if (i == j)
+            (*_prop_d2g[i][j])[_qp] +=  _W_ij[_num_eta * i + j] * 2 * (*_eta[j])[_qp] * (*_eta[j])[_qp];
```

Comment:
I'm not sure this is correct. Both nested loops go from 0 to _num_eta and you acces ```_prop_d2g[i][j]. **However*, please note that ```_prop_d2g[i][j]``` and  ```_prop_d2g[j][i]``` point to the exact same memory location. So you are probably overcounbting here.